Lib/test/test_urlparse.py

```diff
@@ -936,6 +936,16 @@ def test_issue14072(self):
         self.assertEqual(p2.scheme, 'tel')
         self.assertEqual(p2.path, '+31641044153')
 
+    def test_port_casting_failure_message(self):
+        # Assert ValueError when int(string) is parsed as a port value
+        # Asset that the error message is:
+        # port oracle could not be cast to integer value
+        p1 = urllib.parse.urlparse('http://Server=sde; Service=sde:oracle')
+        with self.assertRaisesRegex(ValueError, "Port could not be " \
```

Comment:
Style nit: It's better store the exception message in a `msg` or `message` variable.